### PR TITLE
Fixes to use new ephemeris functions

### PR DESCRIPTION
**Pull request by ljbade**
Needed for https://github.com/swift-nav/libswiftnav/pull/255

### CODE REVIEW

File: src/manage.c [hunk 1]
```diff
@@ -497,9 +497,9 @@ static void manage_track()
 
     acq_status_t *acq = &acq_status[sid_to_index(ch->sid)];
 
-    /* Is ephemeris marked unhealthy? */
+    /* Is ephemeris or alert flag marked unhealthy?*/
     const ephemeris_t *e = ephemeris_get(ch->sid);
-    if (e->valid && !e->healthy) {
+    if (ch->nav_msg.alert || (e->valid && !e->healthy)) {
```
Comment (gsmcmullin) [inline src/manage.c]: This alert flag should probably also be passed through and checked before using a sat in a solution.  This function is called periodically to drop bad sats, but it is independent of the solution code.

Comment (ljbade) [inline src/manage.c]: Ah thanks for explaining that. I thought I needed something in the solution, but I wasn't sure after I saw this line!

I will try to figure out what is needed to change in there.

**Comment by ljbade** [comment]
@gsmcmullin Can you help me with figuring out how to get the alert flags to the solution?

I can't figure out how stuff gets from `trac.c` where the `nav_msg_t` can get got via `tracking_channel_t` to `solution.c` where I can somewhere (or in `pvt.c` from libswiftnav) filter out unhealthy sats (both from alert flag and ephemeris).

**Comment by ljbade** [comment]
@gsmcmullin I manage to figure it out. It gets tricky trying to follow some of the flow of data in the firmware.

I had to make some more changes to libswiftnav so I will make another PR over there for review.

**Comment by ljbade** [comment]
Oops, just realised I made a mistake I didn't pickup in bench test.

**Comment by ljbade** [comment]
@gsmcmullin OK I have created https://github.com/swift-nav/libswiftnav/issues/257, and a new commit here which should exclude unhealthy sats from the solution.

**Comment by ljbade** [comment]
Related issues for my reference: https://github.com/swift-nav/piksi_firmware/issues/550 https://github.com/swift-nav/piksi_firmware/issues/74 https://github.com/swift-nav/piksi_firmware/issues/477 https://github.com/swift-nav/piksi_firmware/issues/475

**Comment by ljbade** [comment]
After chat with @fnoble yesterday going to:
- back out extra struct fields added in https://github.com/swift-nav/libswiftnav/pull/257
- create a `ephemeris_healthy`
- update https://github.com/swift-nav/piksi_firmware/blob/de6b0a65549c8ed6300daf4f8636c1065ef065e0/src/manage.c#L584 to check healthy and good
- review the places `ephemeris_good` is used

**Comment by ljbade** [comment]
@swift-nav/firmware This PR is now ready for HITL testing.

I had to roll https://github.com/swift-nav/piksi_firmware/pull/594 into this PR due to inter-linked changes in libswiftnav.

File: src/manage.c [hunk 2]
```diff
@@ -196,7 +196,7 @@ void manage_acq_setup()
  *  from ephemeris or almanac, if available and elevation > mask
  * \return Score (higher is better)
  */
-static u16 manage_warm_start(gnss_signal_t sid, gps_time_t t,
+static u16 manage_warm_start(gnss_signal_t sid, gps_time_t* t,
```
Comment (jacobmcnamee) [inline src/manage.c]: const ?

**Comment by jacobmcnamee** [comment]
lgtm

File: src/manage.c (hunk 2 above)
Comment (ljbade) [inline src/manage.c]: good catch

**Comment by ljbade** [comment]
I have encountered a bug in HITL testing where this PR marks all sats as unhealthy.

**Comment by jacobmcnamee** [comment]
Update the libswiftnav pointer and I think it's ready.

**Comment by ljbade** [comment]
@jacobmcnamee I updated the pointer. Let me know if good to merge.

**Comment by jacobmcnamee** [comment]
Lgtm. Please merge.

**Pull request merged**